targets/TARGET_NORDIC/TARGET_NRF5/sleep.c

```diff
@@ -75,6 +77,15 @@ void hal_sleep(void)
 
 void hal_deepsleep(void)
 {
+    if (us_ticker_is_enabled()) {
+        /* Disable timer used for us ticker. */
+        nrf_timer_task_trigger(NRF_TIMER1, NRF_TIMER_TASK_STOP);
+    }
+
     hal_sleep();
     //   NRF_POWER->SYSTEMOFF=1;
+
+    if (us_ticker_is_enabled()) {
```

Comment:
Please acquire locally the state of the ticker at the function entry then use this state to decide if the ticker should be enable or not. A sane implementation of `us_ticker_is_enabled` would always return false at this point.